### PR TITLE
[All hosts] (auth) Update supported hosts for NAA

### PR DESCRIPTION
**Pull request by samantharamon**
|        Q        |                    A                    |
| --------------- | --------------------------------------- |
| Bug fix?        | no                              |
| New feature?    | no                          |
| New sample?     | no                       |
| Related issues? | none |

## What's in this Pull Request?

- Updates supported hosts based on [NAA supported accounts and hosts](https://learn.microsoft.com/office/dev/add-ins/develop/enable-nested-app-authentication-in-your-add-in#naa-supported-accounts-and-hosts).
- Adds the NAA samples to the README.
- Updates the table of auth samples in the README to match [Office Add-in code samples](https://learn.microsoft.com/office/dev/add-ins/overview/office-add-in-code-samples#authentication-authorization-and-single-sign-on-sso).

Related PR: https://github.com/OfficeDev/office-js-docs-pr/pull/4714

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,19 +31,21 @@ The following samples are the completed versions of various tutorials for Office
...
+
+The following samples show how to access and work with a user's Microsoft Graph data using the [Microsoft identity platform](https://learn.microsoft.com/entra/identity-platform/v2-overview).
+
+| Name                | Description         |
+|:--------------------|:--------------------|
+| [Office Add-in with SSO using nested app authentication](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/auth/Office-Add-in-SSO-NAA) | Shows how to use MSAL.js nested app authentication (NAA) in an Office Add-in to access Microsoft Graph APIs for the signed-in user. The sample displays the signed-in user's name and email. It also inserts the names of files from the user's Microsoft OneDrive account into the document. |
+| [Outlook add-in with SSO using nested app authentication](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/auth/Outlook-Add-in-SSO-NAA) | Shows how to use MSAL.js nested app authentication (NAA) in an Outlook Add-in to access Microsoft Graph APIs for the signed-in user. The sample displays the signed-in user's name and email. It also inserts the names of files from the user's Microsoft OneDrive account into a new message body. |
```
Comment (AlexJerabek) [inline README.md:41]: Feels awkward to have "Office Add-in" and "Outlook Add-in" be distinct here. Should we add an "Applies to" table column for the host?

File: README.md [hunk 2]
```diff
@@ -31,19 +31,21 @@ The following samples are the completed versions of various tutorials for Office
...
+| [Use SSO with event-based activation in an Outlook add-in](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/auth/Outlook-Add-in-SSO-events) | Shows how to use SSO to access a user's Microsoft Graph data from an event handled by an Outlook add-in. |
+| [Single Sign-on (SSO) Sample Outlook Add-in](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/auth/Outlook-Add-in-SSO) | Uses Office's SSO feature to give the add-in access to Microsoft Graph data. |
```
Comment (AlexJerabek) [inline README.md:43]: We define "SSO" here and in two other rows, but not elsewhere. The first instance on the page should have this.

File: README.md [hunk 3]
```diff
@@ -31,19 +31,21 @@ The following samples are the completed versions of various tutorials for Office
...
-| Excel add-in | MSAL | ASP.NET server |[Use MSAL.NET for auth and Microsoft Graph in an Excel add-in](Samples/auth/Office-Add-in-Microsoft-Graph-ASPNET)|
-| Outlook add-in | MSAL | ASP.NET server |[Use MSAL.NET for auth and Microsoft Graph in an Outlook add-in](Samples/auth/Outlook-Add-in-Microsoft-Graph-ASPNET)|
+## Authentication, authorization, and single sign-on (SSO)
+
+The following samples show how to access and work with a user's Microsoft Graph data using the [Microsoft identity platform](https://learn.microsoft.com/entra/identity-platform/v2-overview).
+
+| Name                | Description         |
+|:--------------------|:--------------------|
+| [Office Add-in with SSO using nested app authentication](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/auth/Office-Add-in-SSO-NAA) | Shows how to use MSAL.js nested app authentication (NAA) in an Office Add-in to access Microsoft Graph APIs for the signed-in user. The sample displays the signed-in user's name and email. It also inserts the names of files from the user's Microsoft OneDrive account into the document. |
```
Comment (AlexJerabek) [inline README.md:40]: What's the rationale of the table order?

**Review by AlexJerabek — APPROVED** [review]
Approving with some table suggestions.

File: README.md (hunk 1 above)
Comment (samantharamon) [reply README.md:41]: Added.

File: README.md (hunk 3 above)
Comment (samantharamon) [reply README.md:40]: To be honest, I only followed the order in [Office Add-in code samples](https://learn.microsoft.com/office/dev/add-ins/overview/office-add-in-code-samples#authentication-authorization-and-single-sign-on-sso). @davidchesnut may have more insight on this.

File: README.md [hunk 4]
```diff
@@ -31,19 +31,21 @@ The following samples are the completed versions of various tutorials for Office
...
-| Excel add-in | MSAL | ASP.NET server |[Use MSAL.NET for auth and Microsoft Graph in an Excel add-in](Samples/auth/Office-Add-in-Microsoft-Graph-ASPNET)|
-| Outlook add-in | MSAL | ASP.NET server |[Use MSAL.NET for auth and Microsoft Graph in an Outlook add-in](Samples/auth/Outlook-Add-in-Microsoft-Graph-ASPNET)|
+## Authentication, authorization, and single sign-on (SSO)
+
+The following samples show how to access and work with a user's Microsoft Graph data using the [Microsoft identity platform](https://learn.microsoft.com/entra/identity-platform/v2-overview).
+
+| Name                | Applies to | Description         |
+|:--------------------|:-----------|:--------------------|
+| [Office Add-in with SSO using nested app authentication](https://github.com/OfficeDev/Office-Add-in-samples/tree/main/Samples/auth/Office-Add-in-SSO-NAA) | - Excel<br>- PowerPoint<br>- Word | Shows how to use MSAL.js nested app authentication (NAA) in an Office Add-in to access Microsoft Graph APIs for the signed-in user. The sample displays the signed-in user's name and email. It also inserts the names of files from the user's Microsoft OneDrive account into the document. |
```
Comment (AlexJerabek) [inline README.md:40]: Not sure if this goes against the style guidelines, but I think the applies to column would look better without the dashes (just the breaks). Just my personal preference to match the single item entries.

Comment (samantharamon) [reply README.md:40]: I couldn't find anything about this in the style guide, so I'll remove the dashes. As you mentioned, it'll make it consistent with the single item entries.

**Pull request merged**